### ISSUE TITLE
Reproducing Differential Transformer.

### ISSUE THREAD

**Issue by Zcchill** [microsoft/unilm#1637]
I compared the training losses of the Transformer and Differential Transformer under my own setting but failed to reproduce the experimental results. I wonder whether the operation of "taking difference" and applying softmax should be reversed in official code.
[official code](https://github.com/microsoft/unilm/blob/095574c8bf5999526df8a069b36b9d96e1fd337e/Diff-Transformer/multihead_diffattn.py#L101)
@YTianZHU

**Comment by YTianZHU** [comment]
Hi, the operation of taking difference and applying softmax is correct in the code. Could you provide a code snippet of your implementation? And you can also provide the model setting in your training.

**Comment by YTianZHU** [comment]
@Zcchill Seems the implementation is correct. The model has 4 heads for Diff, and 8 heads for Transformer, right? Would you provide some description about the results, for example, loss curve comparison, gradient norm comparison, etc.
